### ISSUE TITLE
Screen readers read the collapse/expand icons in hierarchical Grid as link

### ISSUE THREAD

**Issue by nelito987** [telerik/kendo-ui-core#7263]
### Bug report

When the user navigates to an expand or collapse icon to activate it the screen readers are reading it as a 'link'

### Reproduction of the problem
1.Open URL Kendo UI Snippet | [Kendo UI Dojo (telerik.com)](https://dojo.telerik.com/@NeliKondova/eyAVimEN) in browser.
4. Navigate to the grid table UI present inside the right pane.
5. Activate the 'expand' icon and listen to the reader.

### Current behavior

The screen reader is announcing 'link collapse' - [screencast](https://www.screencast.com/t/MdzCkLkgCtua)
Incorrect role defined as "Link" for all the "collapsed/expand" button in the Scoring page. The screen reader announces "enter table 3 by 9 link collapsed, table header Traces".

### Expected/desired behavior

The correct role should be defined as a button for all the "collapsed/expand" button. Screen reader should announce like "enter table 3 by 9 collapsed, table header Traces".

### Environment

* **Kendo UI version:** 2023.1.314
* **Browser:** [all ]

**Comment by Dimitar-Goshev** [comment]
Tested with Kendo UI version 2023.3.1114 - https://dojo.telerik.com/@Dimitar-Goshev/ORAfesAG
The toggle icon anchor element has `aria-hidden="true"` attribute set and `tabindex="-1"` . Thus, the icon is no longer read as a link. Grid's row state is read properly on keyboard navigation, which is available when setting Grid's [navigatable](https://docs.telerik.com/kendo-ui/api/javascript/ui/grid/configuration/navigatable) property to **true** as in the above Dojo.
